### ISSUE TITLE
Flake inputs as sources of crate dependencies/version information

### ISSUE THREAD

**Issue by RichardWarfield** [ipetkov/crane#557]
tldr: is it possible to use a flake input as a vendored crate dependency or otherwise as a source of version information?  There are circumstances where a rust library is implicitly depended on through a flake input and it would be nice to have the flake.lock be the single source of version truth.

(I saw something related was asked [here](https://github.com/ipetkov/crane/issues/149) but that seemed to be specifically about vendoring a library that was missing Cargo.lock entries)

We have several packages that consist of a rust library with a Haskell wrapper.  These are packaged up as a flake, a package output for each library (Haskell+Rust) and a nixpkgs overlay ([haskell.nix](https://github.com/input-output-hk/haskell.nix) expects dependencies to be provided in pkgs, via overlays).

Some downstream consumers need both the Rust package and the overlay and therefore add the repository both as a dependency in Cargo.toml and as an input in their flake.nix.  The problem is, there are now two sources of version information for the rust library: Cargo.lock and the flake.lock, and these can get out of sync.

**Comment by ipetkov** [comment]
> tldr: is it possible to use a flake input as a vendored crate dependency or otherwise as a source of version information?

Technically yes, with some effort: [`vendorCargoDeps` accepts overrides for `Cargo.lock`](https://github.com/ipetkov/crane/blob/880573f80d09e18a11713f402b9e6172a085449f/docs/API.md#input-attributes-2) so it is technically possible to parse `Cargo.lock` at evaluation time (via `builtins.readFile` and `builtins.fromTOML`), patch it up (again at evaluation time) to replace a crate with a path-based dependency and feed it into `vendorCargoDeps`.

Though if you ever want to develop/build the project outside of Nix (i.e. in your editor) you'd probably need some kind of script to point your `Cargo.lock` to the appropriate sources anyway, so some automation there might be an alternative solution

**Comment by RichardWarfield** [comment]
Thank you.  This sounds like something quite difficult to support in the library, so I'l close this.
